src/main/java/dk/kb/datahandler/facade/DsDatahandlerFacade.java

```diff
@@ -24,26 +31,75 @@
 
 
 public class DsDatahandlerFacade {
-    
+
     private static final Logger log = LoggerFactory.getLogger(DsDatahandlerFacade.class);
 
     /**
-    * Starts a delta OAI harvest job for the target. The job will continue from last timestamp
-    * saved on the file system for that target.
-    * The job will harvest records from the OAI server and ingest them into DS-storage  
-    *  
-    * @param  oaiTargetName the location of the image, relative to the url argument
-    * @return Number of harvested records.
-    */    
+     * Ingest records directly into ds-storage from a zip-file containing multiple files that each is an xml-file with a single record
+     *  
+     * @param  recordBase The recordBase for collection documents. The recordBase must be defined in ds-storage. 
+     * @param is Inputstream. Must be a zip-file containing single files that each is an XML record.
+     * @return Number of successfully ingested records.
+     */    
+    public static Integer ingestFromZipfile(String recordBase, InputStream is) throws Exception {
+        ZipInputStream zis = new ZipInputStream(new BufferedInputStream(is));
+
+        DsStorageApi dsAPI = getDsStorageApiClient();
+        int recordsIngested=0;
+
+        ZipEntry entry = null;
+
+        while ((entry = zis.getNextEntry()) != null) {
+
+            String fileName=entry.getName();
+
+            try {                
+                String record_string = IOUtils.toString(zis, StandardCharsets.UTF_8);           
+                Document record = OaiHarvestClient.sanitizeXml(record_string, null);
+
+                //There are several 'mods:identifier' identifier tags, but the first is the URI always.
+                Node item = record.getElementsByTagName("mods:identifier").item(0);            
+                String identifier =  item.getTextContent();                               
+
+                //Example: urn:uuid:096c9090-717f-11e0-82d7-002185371280
+                identifier=identifier.replaceFirst("urn:uuid:", ""); // Clear this first part from the ID
+                
+                String recordId= recordBase+":"+identifier;
+                log.info("Ingesting record filename from zip:"+fileName +" id:"+identifier);
+                DsRecordDto dsRecord = new DsRecordDto();
+                dsRecord.setId(recordId); 
+                dsRecord.setBase(recordBase);
+                dsRecord.setData(record_string);                                                 
+                dsAPI.recordPost(dsRecord);  
```

Comment:
Not an error, but it might be clearer to use chaining here?
```
DsRecordDto dsRecord = new DsRecordDto().
                        id(recordId).
                        base(recordBase).
                        data(record_string);
```

Reply:
I like the getter/setter pattern. I use above only in the same object class.

Reply:
Fair enough.